From: Kurtis Rader <krader@skepticism.us>
Date: Fri, 1 Nov 2019 20:13:20 -0700
Subject: [PATCH] Fix b_sync.sh test failures on some macOS systems
 When run with malloc debugging enabled fd 3 might be open. Switch to a
 fd that should never be valid; i.e., open.
Modified-By: Anuradha Weeraman <aweeraman@gmail.com>

--- a/src/cmd/ksh93/tests/b_sync.sh
+++ b/src/cmd/ksh93/tests/b_sync.sh
@@ -16,12 +16,17 @@ expect=""
 [[ "$actual" == "$expect" ]] || log_error "sync -s1" "$expect" "$actual"
 
 # ==========
-# sync -s3
+# sync -s 999
 # An invalid file descriptor should fail. We don't verify the errno portion of the message because
-# it can vary across systems.
-actual=$(sync -s3 2>&1)
-expect="sync: fsync(3) failed"
-[[ "$actual" =~ "$expect".* ]] || log_error "sync -s3" "$expect" "$actual"
+# it can vary across systems -- even though it should be EBADF which should be the same numeric
+# value on every system that cares at all about being UNIX compatible.
+#
+# We don't use fd 3 because it might be opened by a debug malloc subsystem. We expect fd 999 to be
+# unused if not invalid.
+log_info "HOME=|$HOME| USER=|$USER|"
+actual=$(sync -s 999 2>&1)
+expect="sync: fsync(999) failed"
+[[ "$actual" =~ "$expect".* ]] || log_error "sync -s999" "$expect" "$actual"
 
 # ==========
 # sync -f
